lib/repositories/message.js

```diff
@@ -394,20 +412,89 @@ function createApi(table, provider, sqlGenerator, dependencies) {
                 });
             });
         });
+    },
+
+    /**
+     * Deletes all messages belonging to the specific mailbox
+     */
+    removeByMailbox: function remove(mailbox) {
+      dependencies.logger.trace('message.removeByMailbox called');
+
+      var query = table
+        .select(table.star())
+        .from(table)
+        .where(table['mailbox_id'].equals(mailbox.getId()))
+        .toQuery();
+      query = provider.forUpdate(query);
+
+      return provider.beginTransaction(true)
+        .then(function(transaction) {
+
+          return transaction.runQuery(query)
+            .then(function(result) {
+              var query = table
+                .delete()
+                .where(table['mailbox_id'].equals(mailbox.getId()))
+                .toQuery();
+
+              var messages = result.rows;
+
+              var x;
+              for (x in messages) {
+                messages[x] = convertFromStorage(messages[x]);
+              }
+
+              return transaction.runQuery(query)
+                .then(function() {
+                  return transaction.commit();
+                })
+                .then(function() {
+                  dependencies.logger.debug({messages: messages},
+                                            'Messages removed');
+                  return messages;
```

Comment:
Why are messages being returned here? The messages won't be fully formed model objects since it doesn't use the common helper methods. The purpose of convertFromStorage is just to get the 'read' property to be converted to a boolean. Not sure this is necessary here. It's used in the message repository to only set the message as read once since it updates MWI and we don't want to duplicate that call.